modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.integration.test.idp.mgt;
+
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.axis2.context.ConfigurationContextFactory;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.IdentityProvider;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.Property;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.ProvisioningConnectorConfig;
+import org.wso2.carbon.integration.common.admin.client.AuthenticatorClient;
+import org.wso2.identity.integration.common.clients.Idp.IdentityProviderMgtServiceClient;
+import org.wso2.identity.integration.common.utils.ISIntegrationTest;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This test class is to test the resident IDP config URLs in super tenant legacy URL mode and tenant legacy URL mode.
+ */
+public class ResidentIDPConfigsTestCase extends ISIntegrationTest {
+
+    private IdentityProviderMgtServiceClient idpMgtServiceClient;
+    private AuthenticatorClient loginManger;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        ConfigurationContext configContext = ConfigurationContextFactory.createConfigurationContextFromFileSystem(null, null);
+        idpMgtServiceClient = new IdentityProviderMgtServiceClient(sessionCookie, backendURL, configContext);
+    }
+
+    @Test(groups = "wso2.is", description = "Test resident IdP config URLs in super tenant mode")
+    public void testResidentIdPConfigs() throws Exception {
+
+        log.info("Retrieving resident identity provide");
```

Comment:
`provide` -> `provider`

Reply:
This code was changed with https://github.com/wso2/product-is/commit/a706b3711c3559a331e7d18c1d30746e787b70f5